phoenix-core/src/main/java/org/apache/phoenix/monitoring/connqueryservice/ConnectionQueryServicesMetricsManager.java

```diff
@@ -0,0 +1,375 @@
+                    }
+                }
+            }
+        }
+        return localRef;
+    }
+
+    private PhoenixConnectionQueryServiceMetrics getConnectionQueryServiceMetricsInstance(
+            String connectionQueryServiceName) {
+        if (!isConnectionQueryServiceMetricsEnabled) {
+            return null;
+        }
+        if (Strings.isNullOrEmpty(connectionQueryServiceName)) {
+            LOGGER.warn("connection query service Name can't be null or empty");
+            return null;
+        }
+
+        PhoenixConnectionQueryServiceMetrics tInstance;
+        tInstance = connectionQueryServiceMetricsMapping.get(connectionQueryServiceName);
+        if (tInstance == null) {
+            synchronized (ConnectionQueryServicesMetricsManager.class) {
+                tInstance = connectionQueryServiceMetricsMapping.get(connectionQueryServiceName);
+                if (tInstance == null) {
+
+                    LOGGER.info(String.format("connection query service metrics creating object"
+                            + " for connection query service: %s", connectionQueryServiceName));
+                    tInstance =
+                            new PhoenixConnectionQueryServiceMetrics(connectionQueryServiceName,
+                                    options.getConfiguration());
+                    connectionQueryServiceMetricsMapping.put(connectionQueryServiceName, tInstance);
+                }
+            }
+        }
+        return tInstance;
+    }
+
+    /**
+     * This function will be used to add individual MetricType to LocalStore. Also this will serve
+     * as LocalStore to store connection query service metrics before their current value is added
+     * to histogram.
+     * This func is only used for metrics which are counter based, where values increases or
+     * decreases frequently. Like Open Conn Counter. This function will first retrieve it's current
+     * value and increment or decrement (by +/-1) it as required then update the new values.
+     * <br>
+     * Example :- OPEN_PHOENIX_CONNECTIONS_COUNTER, OPEN_INTERNAL_PHOENIX_CONNECTIONS_COUNTER
+     * <br>
+     * <br>
+     * histogram will update with each increment/decrement.
+     * @param connectionQueryServiceName
+     * @param type
+     * @param value
+     */
+    private void updateMetricsAsCounter(String connectionQueryServiceName, MetricType type,
+            long value) {
+
+        long startTime = EnvironmentEdgeManager.currentTime();
+
+        PhoenixConnectionQueryServiceMetrics
+                tInstance =
+                getConnectionQueryServiceMetricsInstance(connectionQueryServiceName);
+        if (tInstance == null) {
+            LOGGER.trace("Phoenix connection query service metrics are disabled for connection"
+                    + " query service: " + connectionQueryServiceName);
+            return;
+        }
+        tInstance.changeMetricValue(type, value);
+
+        if (LOGGER.isDebugEnabled()) {
```

Comment:
is isDebugEnabled check required for debug logging? what happens if its not enabled and still you have debug Log?

Reply:
It's better to check for if logging level is enabled or not. It would avoid processing of log message, trace/debug level logging could be happening too frequently. I have also changed the logging level to TRACE since we may not need it outside of dev enviornment.